server/region_syncer/client.go

```diff
@@ -204,9 +204,10 @@ func (s *RegionSyncer) StartSyncWithLeader(addr string) {
 							core.SetWrittenKeys(stats[i].KeysWritten),
 							core.SetReadBytes(stats[i].BytesRead),
 							core.SetReadKeys(stats[i].KeysRead),
+							core.SetFromHeartbeat(false),
 						)
 					} else {
-						region = core.NewRegionInfo(r, regionLeader)
+						region = core.NewRegionInfo(r, regionLeader, core.SetFromHeartbeat(false))
```

Comment:
It seems we do not need to `SetFromHeartbeat(fase)` because we do not process the value if `IsNew` on L218
```
_, saveKV, _, _ := regionGuide(region, origin)
```